frame/support/src/traits.rs

```diff
@@ -1518,6 +1518,15 @@ pub trait OnFinalize<BlockNumber> {
 	fn on_finalize(_n: BlockNumber) {}
 }
 
+#[impl_for_tuples(30)]
+pub trait OnIdle<BlockNumber> {
+	/// The block is being finalized. Implement to have something happen.
+	///
+	/// NOTE: This function is called AFTER ALL extrinsics in a block are applied,
+	/// including inherent extrinsics.
+	fn on_idle(_n: BlockNumber) {}
```

Comment:
I think at the end the function can look like:
```
fn on_idle(remaning_weight: &mut Weight) {
}
```
So that each pallet decrease the remaining_weight from what they use (note you might need an explicit for_tuple implementation, similar to OnInitialize trait)

Reply:
yup makes sense, and is tied to what @gavofyork is saying above, will implement.

Reply:
I would prefer returning the used `Weight` to be consistent with the other hooks. The subtraction should be done in the `for_tuple` impl (see my other comment).